### DISCUSSION TITLE
Modelling a building floor

### DISCUSSION THREAD

**Discussion by nicolas-f** [Ifsttar/I-Simpa#376] [Show and tell]
Modeling in 3D the following main room and corridor in I-Simpa thanks to Inkscape and OpenSCAD:

![plan](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/56cf0643-190a-4407-9afd-e13bbc23b009)

Install Inkscape

Install OpenSCAD

Open Inkscape and drag and drop the image

To calculate the image size so that 48.1 mm in vector corresponds to 481 cm of the plane:

The image is 2108 pixels wide and the living room is 776 pixels wide.

The image should therefore be 2108*48.1/776=130.66 mm wide.

Click on the cursor in the top left-hand corner and select the image, 

click on the lock to lock width height of the image:

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/92ba07f6-6e22-452e-944e-3cad7104e29d)

In width, type 130.66

We're now going to draw the outline of the propagation zone

Click on the line segment tool:

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/fc37389c-a74f-4039-a787-c97b891d87af)

Select the create a sequence of paraxial line segments mode.

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/dbf06aff-17dc-4c2e-b8a5-16c47a93fc2f)

Hold down the Ctrl key and draw the first wall (to align at 90°).

Continue the contour until you click on the first point you made.

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/e1bda1d1-a324-4a27-a240-2361dc40f088)

Select your outline and copy and paste it into a new Inkscape document,

Adjust the page to the size of the object.

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/30921e88-64cb-4e1a-8d21-1a76a184428b)

Save your project as plan.svg

Open openscad and save the empty project in the same folder as plan.svg

In the editor, type :

linear_extrude(2400) scale(100) import("plan.svg");

the plan is imported, resized to actual size (mm unit on openscad) and extruded to a height of 2.4 m.
Click on Design/"Calculate rendering" ( F6 )

![image](https://github.com/Universite-Gustave-Eiffel/I-Simpa/assets/1382241/b2013f6e-9ff6-4c85-a6b5-5af813afb101)

Edit settings of OpenSCAD and check **Advanced/Export Features/Default to ASCII STL export**

Export in STL format

In I-SIMPA, import the model, select **mm** unit then validate.

Your project is ready for the simulation (you can check the 3d model by running a mesh preview)

**Comment by Vinom009** [comment]
Where the doors and windows and chairs and etc. How supposed add them

**Comment by zoneh5** [comment]
I have wish to make for [Wood flooring Edinburgh](https://www.hoffparquet.co.uk/) can you tell me your decision tree
